webapp/components/Panel.js

```diff
@@ -1,13 +1,35 @@
 import React, { PureComponent } from 'react';
-import { Route } from 'react-router-dom';
+import { Route, Switch } from 'react-router-dom';
 import PropTypes from 'prop-types';
 
 import { getRouteProps } from '../plugins/plugins';
 
 export default class extends PureComponent {
   constructor(props) {
     super(props);
-    this.routes = [];
+    const { customChildren = [], paths } = props;
+    this.routes = customChildren.map(({ Component, metadata }) => {
+      const { name } = metadata;
+
+      const pathName = paths[name]; // will be a unique path from state
+      let path;
+
+      try {
+        if (!name) throw 'Must pass a name for custom Panels';
```

Comment:
no need for `new Error`?

Reply:
You can also just throw a statement: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/throw
At some point it would be good to build out a more robust error catching system that can display notifications of some kind.